### PR TITLE
update usage-analysis-page

### PR DESCRIPTION
**Pull request by Leowisd**
This implemented https://github.com/oncokb/oncokb/issues/2419

And please review this PR: https://github.com/knowledgesystems/knowledgesystems-k8s-deployment/pull/105

### CODE REVIEW

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/vm/usageAnalysis/ResourceModel.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
     
     public ResourceModel(String s){
         this.method =  s.split(" ")[0];
-        this.endpoint = s.split(" ")[1];
+        this.endpoint = s.split(" ")[1].replaceAll("\\%20", "");
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/web/rest/vm/usageAnalysis/ResourceModel.java:10]: Replace the space escape to nothing?

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisController.java [hunk 2]
```diff
@@ -127,4 +130,19 @@ private JSONObject requestData(String file)
 
         return new ResponseEntity<UsageSummary>(summary, status);
     }
+
+    @GetMapping("/usage/resource/detail")
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisController.java:134]: Should we just do `/user/resource/{endpoint}`?

File: src/main/webapp/app/config/constants.tsx [hunk 3]
```diff
@@ -78,6 +78,8 @@ export const NOTIFICATION_TIMEOUT_MILLISECONDS = 5000;
 export const TOKEN_ABOUT_2_EXPIRE_NOTICE_IN_DAYS = 14;
 export const USAGE_TOP_USERS_LIMIT = 10000;
 export const USGAE_ALL_TIME_KEY = 'All';
+export const USGAE_ALL_TIME_VALUE = 'One year';
+export const USAGE_DETAIL_TIME_KEY = 'Detail';
```
Comment (zhx828) [inline src/main/webapp/app/config/constants.tsx:82]: Should these two be updated?

File: src/main/webapp/app/pages/usageAnalysisPage/UserUsageDetailsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  defaultTimeType: number;
+};
+
+@observer
+export default class UserUsageDetailsTable extends React.Component<
+  IUserUsageDetailsTable,
+  {}
+> {
+  @observable resourcesTypeToggleValue = this.props.defaultResourcesType;
+  @observable timeTypeToggleValue = this.props.defaultTimeType;
```
Comment (zhx828) [inline src/main/webapp/app/pages/usageAnalysisPage/UserUsageDetailsTable.tsx:28]: We should type this variable. I would not understand what 1 represents and what 2.

**Review by zhx828 — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/pages/usageAnalysisPage/UsageAnalysisPage.tsx [hunk 5]
```diff
@@ -83,15 +124,40 @@ export default class UsageAnalysisPage extends React.Component<{
...
+    }
+
     return (
       <>
-        <Tabs defaultActiveKey="users" id="uncontrolled-tab-example">
-          <Tab eventKey="users" title="Users">
+        <Tabs
+          defaultActiveKey={this.usageType}
+          id="uncontrolled-tab-example"
+          onSelect={k => this.toggleType(UsageType[k || UsageType.USER])}
```
[the remark this answers is not in the document]
Comment (Leowisd) [reply src/main/webapp/app/pages/usageAnalysisPage/UsageAnalysisPage.tsx:201]: The type of k is `string|null`

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/vm/usageAnalysis/ResourceModel.java (hunk 1 above)
Comment (Leowisd) [reply src/main/java/org/mskcc/cbio/oncokb/web/rest/vm/usageAnalysis/ResourceModel.java:10]: The target is removing unexpected space occurred in the endpoints, or sometimes endpoints will be shown like `/api/v1/annotate/%20%20%20%20%20%20%20%20%20%20%20%20%20mutations`

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisController.java (hunk 2 above)
Comment (Leowisd) [reply src/main/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisController.java:134]: '/' in endpoint will cause matching error.

File: src/main/webapp/app/config/constants.tsx (hunk 3 above)
Comment (Leowisd) [reply src/main/webapp/app/config/constants.tsx:82]: I remember we think 'One Year' looks good, and the 'Detail' is not for showing

File: src/main/webapp/app/pages/usageAnalysisPage/UsageAnalysisPage.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (zhx828) [reply src/main/webapp/app/pages/usageAnalysisPage/UsageAnalysisPage.tsx:201]: But you have the defaultActiveKey. Even though k by definition here has the type null, but it should never be null?

File: src/main/webapp/app/pages/usageAnalysisPage/UsageToggleGroup.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { observer } from 'mobx-react';
+import React from 'react';
+import { ToggleButton, ToggleButtonGroup } from 'react-bootstrap';
+import { ToggleValue } from './UsageAnalysisPage';
+
+type IUsageToggleGroupProps = {
+  defaultValue: ToggleValue;
+  toggleValueOne: ToggleValue;
+  toggleValueTwo: ToggleValue;
```
Comment (zhx828) [inline src/main/webapp/app/pages/usageAnalysisPage/UsageToggleGroup.tsx:9]: Should this be an array?

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisController.java (hunk 2 above)
Comment (zhx828) [reply src/main/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisController.java:134]: But I believe the api request has encoded the slash?

File: src/main/webapp/app/pages/usageAnalysisPage/UsageAnalysisPage.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Leowisd) [reply src/main/webapp/app/pages/usageAnalysisPage/UsageAnalysisPage.tsx:201]: It should never be null, but the typescript will give error `Type 'null' cannot be used as an index type`. Any idea to solve it?

[the remark this answers is not in the document]
Comment (zhx828) [reply src/main/webapp/app/pages/usageAnalysisPage/UsageAnalysisPage.tsx:201]: You can force the type using `!`

File: src/main/webapp/app/pages/usageAnalysisPage/ResourceUsageDetailsTable.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+                    ? 'Duration'
+                    : 'Time'}
+                </span>
+              ),
+              minWidth: 100,
+              accessor: 'time',
+              onFilter: (data: UsageRecord, keyword) =>
+                filterByKeyword(data.time, keyword),
+            },
+          ]}
```
Comment (zhx828) [inline src/main/webapp/app/pages/usageAnalysisPage/ResourceUsageDetailsTable.tsx:82]: For the duplicated column definitions here and UserUsageTable, could you consolidate them?

Comment (Leowisd) [reply src/main/webapp/app/pages/usageAnalysisPage/ResourceUsageDetailsTable.tsx:82]: I think the column definition with `id: 'resource'` in those two tables are different. And the time column also related to each `timeTypeToggleValue`. Do you think it is still necessary to consolidate them?

Comment (zhx828) [reply src/main/webapp/app/pages/usageAnalysisPage/ResourceUsageDetailsTable.tsx:82]: Yup, I think we should considering they are both using UsageRecord model as data. They could just be defaults and you can overwrite properties at the place of calling.

Comment (Leowisd) [reply src/main/webapp/app/pages/usageAnalysisPage/ResourceUsageDetailsTable.tsx]: @zhx828  Sure, have updated

**Review by zhx828 — APPROVED** [review]

**Pull request merged**